Ginger/Ginger/Activities/ActivitiesGroupPage.xaml.cs

```diff
@@ -76,7 +79,131 @@ public ActivitiesGroupPage(ActivitiesGroup activitiesGroup, BusinessFlow parentB
                 SharedRepoInstanceUC.Visibility = Visibility.Collapsed;
                 SharedRepoInstanceUC_Col.Width = new GridLength(0);
             }
-        }     
+        }
+
+        void UpdateSharedRepositorySupportedOperations()
+        {
+            grdGroupedActivities.ShowUpDown = Visibility.Visible;
+
+            grdGroupedActivities.AddToolbarTool(Amdocs.Ginger.Common.Enums.eImageType.Add, "Add Another Activity to this Group", BtnAdd_Click);
+            grdGroupedActivities.AddToolbarTool(Amdocs.Ginger.Common.Enums.eImageType.DeleteSingle, "Delete Current Selected Activity from this Group", GroupActivitiesHandler);
+        }
+
+        private void BtnDelete_Click(object sender, RoutedEventArgs e)
+        {
+            FrameworkElement fElem = sender as FrameworkElement;
+            if (fElem != null)
+            {
+
+            }
+        }
+
+        private void BtnAdd_Click(object sender, RoutedEventArgs e)
+        {
+            ActivitiesRepositoryPage ActivitiesRepoPage = new ActivitiesRepositoryPage(WorkSpace.Instance.SolutionRepository.GetRepositoryItemRootFolder<Activity>(), null, null, GroupActivitiesHandler, true);
+            ActivitiesRepoPage.ShowAsWindow(Window.GetWindow(this));
+        }
+
+        private void GroupActivitiesHandler(object sender, RoutedEventArgs e)
+        {
+            ucGrid senderGrid = sender as ucGrid;
+
+            ObservableList<Amdocs.Ginger.Repository.RepositoryItemBase> selectedActivitiesList = null;
+            if (senderGrid != null)
+            {
+                selectedActivitiesList = senderGrid.GetSelectedItems();
+            }
+            else
+            {
+                selectedActivitiesList = new ObservableList<Amdocs.Ginger.Repository.RepositoryItemBase>();
+                foreach(ActivityIdentifiers actID in grdGroupedActivities.GetSelectedItems())
+                {
+                    selectedActivitiesList.Add(actID.IdentifiedActivity);
+                }
+            }
+
+            if (selectedActivitiesList != null && selectedActivitiesList.Count > 0)
+            {
+                foreach (Activity sharedActivity in selectedActivitiesList)
+                {
+                    if (senderGrid != null)
+                    {
+                        Activity newInstance = sharedActivity.CreateInstance(true) as Activity;
+                        mActivitiesGroup.AddActivityToGroup(newInstance);
+                    }
+                    else
+                    {
+                        mActivitiesGroup.RemoveActivityFromGroup(sharedActivity);
+                    }
+                }
+            }
+        }
+
+        void ShiftItem(bool ShiftUp)
```

Comment:
why you needed to implement the shift up/down?
it already done genericlt by ucGrid, no?

Reply:
Done !